src/iden3comm/packers/anon-crypt.ts

```diff
@@ -0,0 +1,243 @@
+import { Resolvable } from 'did-resolver';
+import { JoseService } from '../../kms/services/jose.service';
+import { base58ToBytes, base64UrlToBytes, byteDecoder, byteEncoder, hexToBytes } from '../../utils';
+import {
+  AcceptJweKEKAlgorithms,
+  MediaType,
+  ProtocolVersion,
+  VerificationMethodType
+} from '../constants';
+import { BasicMessage, DIDDocument, IPacker, VerificationMethod } from '../types';
+import { parseAcceptProfile, resolveVerificationMethods } from '../utils';
+import { KMS, KmsKeyId } from '../../kms';
+import { DID } from '@iden3/js-iden3-core';
+
+export type RecipientInfo = {
+  did: DID;
+  didDocument?: DIDDocument;
+  keyType?: VerificationMethodType;
+  alg?: AcceptJweKEKAlgorithms;
+};
+
+export type JWEPackerParams = {
+  enc: string;
+  recipients: RecipientInfo[];
+};
+
+export class AnonCryptPacker implements IPacker {
+  private readonly _supportedProtocolVersions = [ProtocolVersion.V1];
+
+  private _supportedAlgorithms = [
+    AcceptJweKEKAlgorithms.RSA_OAEP_256,
+    AcceptJweKEKAlgorithms.ECDH_ES_A256KW
+  ];
+
+  constructor(
+    private readonly _joseService: JoseService,
+    private readonly _kms: KMS,
+    private readonly _documentResolver: Resolvable,
+    private _kmsKeyIds: KmsKeyId[]
+  ) {}
+
+  packMessage(msg: BasicMessage, param: JWEPackerParams): Promise<Uint8Array> {
+    return this.packInternal(msg, param);
+  }
+
+  getSupportedKeyIds(): KmsKeyId[] {
+    return this._kmsKeyIds;
+  }
+
+  registerSupportedKeyIds(keyIds: KmsKeyId[]): void {
+    this._kmsKeyIds = [...new Set([...this._kmsKeyIds, ...keyIds])];
+  }
+
+  getSupportedAlgorithms(): AcceptJweKEKAlgorithms[] {
+    return this._supportedAlgorithms;
+  }
+
+  registerSupportedAlgorithm(algorithm: AcceptJweKEKAlgorithms): void {
+    this._supportedAlgorithms = [...new Set([...this._supportedAlgorithms, algorithm])];
+  }
+
+  /**
+   * creates JSON Web Signature token
+   *
+   * @param {Uint8Array} payload - serialized message
+   * @param {PackerParams} params - sender id and proving alg are required
+   * @returns `Promise<Uint8Array>`
+   */
+  async pack(payload: Uint8Array, params: JWEPackerParams): Promise<Uint8Array> {
+    const message = JSON.parse(byteDecoder.decode(payload));
+    return this.packInternal(message, params);
+  }
+
+  async unpack(envelope: Uint8Array): Promise<BasicMessage> {
+    const jwe = JSON.parse(byteDecoder.decode(envelope));
+
+    const promises = this._kmsKeyIds.map((keyId) =>
```

Comment:
We need to check recipients of jwe and decrypt with a key that can can be resolved by specific recipient. E.g recipient is did:123 getKmsKeyId(did) should be passed. Implementation function can use resolved ddid document of the recipient to find key id or public key which can lead to kns key id